src/main/java/teammates/ui/controller/AdminSessionsPageAction.java

```diff
@@ -30,141 +29,116 @@
     private int totalClosedStatusSessions;
     private int totalWaitToOpenStatusSessions;
     private int totalInstitutes;
-    private Date rangeStart;
-    private Date rangeEnd;
-    private double zone;
+    private LocalDateTime rangeStart;
+    private LocalDateTime rangeEnd;
+    private ZoneId timeZone;
     private boolean isShowAll;
 
     @Override
     protected ActionResult execute() {
-
         gateKeeper.verifyAdminPrivileges(account);
         data = new AdminSessionsPageData(account, sessionToken);
-
         isShowAll = getRequestParamAsBoolean("all");
 
-        ActionResult result = createShowPageResultIfParametersInvalid();
+        ActionResult result = validateParametersAndCreateShowPageResultIfInvalid();
         if (result != null) {
             return result;
         }
 
-        Date rangeStartUtc = TimeHelper.convertLocalDateToUtc(rangeStart, zone);
-        Date rangeEndUtc = TimeHelper.convertLocalDateToUtc(rangeEnd, zone);
-        List<FeedbackSessionAttributes> allOpenFeedbackSessionsList =
-                logic.getAllOpenFeedbackSessions(rangeStartUtc, rangeEndUtc);
+        List<FeedbackSessionAttributes> allOpenFeedbackSessionsList = logic.getAllOpenFeedbackSessions(
+                TimeHelper.convertLocalDateTimeToInstant(rangeStart, timeZone),
+                TimeHelper.convertLocalDateTimeToInstant(rangeEnd, timeZone)
```

Comment:
Note that this change will change behaviour, as `convertLocalDateToUtc` will not change special time. However, special time doesn't have any meaning in the search query, so this is ok.